src/utils/string/StringUtils.js

```diff
@@ -576,6 +576,18 @@ const getOfficialAnonUsername = (anonUserInfo) => {
   return `${colorName} ${emojiName}`;
 };
 
+const handleUserName = (item, selectedChannel) => {
+  if (item?.user?.anon_user_info_emoji_code) {
+    return getOfficialAnonUsername(item?.user);
+  }
+
+  if (item?.user?.username !== 'AnonymousUser') {
+    return item?.user?.username;
+  }
+
+  return getOfficialAnonUsername(selectedChannel);
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `handleUserName` seems to be handling different cases for user names. However, it's not clear what the expected behavior is when `item?.user?.username === 'AnonymousUser'`. In this case, the function returns `getOfficialAnonUsername(selectedChannel)`, but it's unclear why the selected channel's official anonymous username is returned instead of the item's user's anonymous username. Please ensure that this is the intended behavior.

```javascript
const handleUserName = (item, selectedChannel) => {
  if (item?.user?.anon_user_info_emoji_code) {
    return getOfficialAnonUsername(item?.user);
  }

  if (item?.user?.username !== 'AnonymousUser') {
    return item?.user?.username;
  }

  return getOfficialAnonUsername(selectedChannel);
};
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->